Files/Helpers/JumpListManager.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Windows.UI.StartScreen;
+using Windows.Storage;
+using Files.Common;
+
+namespace Files.Helpers
+{
+    public sealed class JumpListManager
+    {
+        private JumpList _instance = null;
+        private List<string> JumpListItemPaths { get; set; }
+        public JumpListManager()
+        {
+            Initialize();
+        }
+
+        private async void Initialize()
+        {
+            if (JumpList.IsSupported())
+            {
+                _instance = await JumpList.LoadCurrentAsync();
+
+                // Disable automatic jumplist. It doesn't work with Files UWP.
+                _instance.SystemGroupKind = JumpListSystemGroupKind.None;
+                JumpListItemPaths = _instance.Items.Select(item => item.Arguments).ToList();
+            }
+        }
+
+        public async void AddFolderToJumpList(string path)
+        {
+            await AddFolder(path);
+            await _instance?.SaveAsync();
+        }
+
+        private Task AddFolder(string path)
+        {
+            if (!JumpListItemPaths.Contains(path) && _instance != null)
+            {
+                string displayName;
+                if (path.Equals(App.AppSettings.DesktopPath, StringComparison.OrdinalIgnoreCase))
+                {
+                    displayName = "ms-resource:///Resources/SidebarDesktop";
+                }
+                else if (path.Equals(App.AppSettings.DownloadsPath, StringComparison.OrdinalIgnoreCase))
+                {
+                    displayName = "ms-resource:///Resources/SidebarDownloads";
+                }
+                else if (path.Equals(App.AppSettings.DocumentsPath, StringComparison.OrdinalIgnoreCase))
+                {
+                    displayName = "ms-resource:///Resources/SidebarDocuments";
+                }
+                else if (path.Equals(App.AppSettings.PicturesPath, StringComparison.OrdinalIgnoreCase))
+                {
+                    displayName = "ms-resource:///Resources/SidebarPictures";
+                }
+                else if (path.Equals(App.AppSettings.MusicPath, StringComparison.OrdinalIgnoreCase))
+                {
+                    displayName = "ms-resource:///Resources/SidebarMusic";
+                }
+                else if (path.Equals(App.AppSettings.VideosPath, StringComparison.OrdinalIgnoreCase))
+                {
+                    displayName = "ms-resource:///Resources/SidebarVideos";
```

Comment:
You can use `ResourceController.GetTranslation("SidebarVideos")` instead of this

Reply:
I know however when you assign the ms-resource:/// protocol to a jumplist item, Windows will automatically retrieve the string for the current language. This way we don't need to refresh the jumplist after the user changes their language. However we need to refresh the jumplist after an update since the ms-resource:/// protocol is version dependent.